### ISSUE TITLE
[BUG]: Long production DFP training times

### ISSUE THREAD

**Issue by efajardo-nv** [nv-morpheus/Morpheus#816]
### Version

23.01

### Which installation method(s) does this occur on?

Source

### Describe the bug.

We are experiencing long training times with our production DFP pipeline. There appears to be an issue with the `dfencoder` AutoEncoder not fully utilizing our GPU. We're seeing about two hours on first epoch with a dataset that looks like this:

- 7 categorical variables with cardinality of: 854, 17, 17, 989, 989, 67, and 7455.
- 800 binary features.
- 100-200M rows of data in total

### Code of Conduct

- [X] I agree to follow Morpheus' Code of Conduct
- [X] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

**Comment by efajardo-nv** [comment]
Confirmed the low GPU utilization during training. Most of the training time is spent during the post-processing phase of the `fit` call, specifically in `get_anomaly_score_losses`.  By default, `get_anomaly_score_losses` uses entire training dataframe for calculating loss statistics. By setting `use_val_for_loss_stats=True` in `fit` call, we can use much smaller validation set instead. This resulted in 2.5x speedup in my training run using `validation_size=0.1`.

For added speedup, `get_anomaly_score_losses in `autoencoder.py` can be updated to use self.eval_batch_size instead of self.batch_size (#861). This will default to use 1024 instead of 512. This resulted in 3.2x speedup when combined with use_val_for_loss_stats=True .

**Comment by jarmak-nv** [comment]
@efajardo-nv should this still be open? Just want to make sure #861 didn't close this out.

**Comment by efajardo-nv** [comment]
@jarmak-nv yes this issue is still ongoing. #861 did not provide enough training improvement. waiting for feedback on the following suggestions:
- training single batch using new `DFPFileBatcherStage.period == None` option provided in PR #912 (instead of multiple incremental models).
- downsampling using improved sampling options also provided in PR #912 to reduce training data size
- larger training batch size

**Comment by jarmak-nv** [comment]
Got it, thanks for the update! 

(Now that I'm fully in the issue instead of from the project view I see the mention of other issues in the timeline. Not the first time the lack of timeline from the project view has gotten me, and probably not the last. Apologies!)

**Comment by efajardo-nv** [comment]
After incorporating all the above suggestions, their training time still was not significantly improved. It was also found that training outside of the pipeline (i.e. using `dfencoder` directly) is around 10x faster compared to running inside the pipeline. 

I was able to reproduce this discrepancy using 23.03 with the 23.07 updates corresponding to the above fixes. The discrepancy appears to go away when I update to 23.07.  Morpheus update is in progress now. Waiting on feedback

**Comment by efajardo-nv** [comment]
Issue resolved by migrating to 23.07 environment and pipeline.
